tests/validator/test_validator.py

```diff
@@ -447,3 +453,104 @@ def test_head(
         "SexCode": {0: 1, 1: 1, 2: 0, 3: 1, 4: 0},
     }
     assert head.to_dict() == expected
+
+
+@pytest.fixture()
+def multi_batch_taxi_validator(yellow_trip_pandas_data_context):
+    context = yellow_trip_pandas_data_context
+
+    suite = context.create_expectation_suite("validating_taxi_data")
+
+    multi_batch_request = BatchRequest(
+        datasource_name="taxi_pandas",
+        data_connector_name="monthly",
+        data_asset_name="my_reports",
+        data_connector_query={"batch_filter_parameters": {"year": "2019"}},
+    )
+
+    validator_multi_batch = context.get_validator(
+        batch_request=multi_batch_request, expectation_suite=suite
+    )
+
+    return validator_multi_batch
+
+
+def test_validator_can_instantiate_with_a_multi_batch_request(
+    multi_batch_taxi_validator,
+):
+    assert len(multi_batch_taxi_validator.batches) == 3
+    assert (
+        multi_batch_taxi_validator.active_batch.batch_definition.batch_identifiers[
+            "month"
+        ]
+        == "03"
+    )
+
+    validator_batch_identifiers_for_all_batches = [
+        i for i in multi_batch_taxi_validator.batches
+    ]
+    assert validator_batch_identifiers_for_all_batches == [
+        "18653cbf8fb5baf5fbbc5ed95f9ee94d",
+        "92bcffc67c34a1c9a67e0062ed4a9529",
+        "021563e94d7866f395288f6e306aed9b",
+    ]
+
+
+def test_validator_batch_filter(multi_batch_taxi_validator):
+    total_batch_definition_list = [
```

Comment:
@talagluck Type hints, please. :). Also, I did not see a test involving a custom function... If you do not mind, might as well include one for the sake of completeness (and also to serve as a nice usage example).  Thanks!

Reply:
Done and done!